### PR TITLE
Fixed Inactivity bug, refined bullet collision and added server comma…

### PR DESCRIPTION
**Pull request by Ploug**
…nds.

Why have the InactivityEvent, when ping is working?

### CODE REVIEW

File: Ball of Duty Server/Program.cs [hunk 1]
```diff
@@ -13,25 +14,63 @@ namespace Ball_of_Duty_Server
...
+            }
+            else
+            {
+                return null;
+            }
+        }
+
+        public static string GetCommand(String line)
+        {
+            return line.IndexOf(" ") < 0 ? "" : line.Substring(line.IndexOf(" ") + 1);
```
Comment (LuqJensen) [inline Ball of Duty Server/Program.cs:50]: ?
line.Split(' ');?

File: Ball of Duty Server/Program.cs [hunk 2]
```diff
@@ -13,25 +14,63 @@ namespace Ball_of_Duty_Server
...
+            {
+                return Command.WRITE_ACTIVE_PLAYERS;
+            }
+            else if (line.ToLower().StartsWith("ban"))
+            {
+                return Command.BAN;
+            }
+            else
+            {
+                return null;
```
Comment (LuqJensen) [inline Ball of Duty Server/Program.cs]: you should always define a default value in an enum so you avoid using nullables or other silly stuff for enum values.
Command.NONE = 0

File: Ball of Duty Server/Program.cs [hunk 3]
```diff
@@ -53,9 +95,34 @@ private static void Main(string[] args)
...
+                        case Command.WRITE_ACTIVE_PLAYERS:
+                        {
+                            break; // TODO
+                        }
+                        case Command.BAN:
+                        {
+                            break; // TODO
+                        }
+                    }
+                    lineRead = Console.ReadLine();
```
Comment (LuqJensen) [inline Ball of Duty Server/Program.cs]: do while?

File: Ball of Duty Server/Domain/Communication/Broker.Translate.cs [hunk 4]
```diff
@@ -19,9 +19,10 @@ private void AddOpcodeMapping()
         {
             _opcodeMapping.Add(Opcodes.POSITION_UPDATE, this.ReadPositionUpdate);
             _opcodeMapping.Add(Opcodes.REQUEST_BULLET, this.BulletCreationRequest);
-            _opcodeMapping.Add(Opcodes.PING, br => { br.ReadByte(); }); // just read ASCII.EOT
+            _opcodeMapping.Add(Opcodes.PING, br => { br.Read(); }); // just read ASCII.EOT
```
Comment (LuqJensen) [inline Ball of Duty Server/Domain/Communication/Broker.Translate.cs]: ?

**Comment by LuqJensen** [comment]
you should ask for the reason before attempting to remove it. 
InactivityEvent and TimeoutEvent do not serve the same purpose.
TimeoutEvent is made to detect when a client disconnects due to timeouts.
InactivityEvent is made to determine whether a client is afking for too long. 
You see, if a client has a stable TCP connection to the server, he can remain forever online and consume a spot in the game despite not playing. TimeoutEvent wont prevent that, it is not the purpose of that event. Thats why we have an InactivityEvent aswell.

File: Ball of Duty Server/Program.cs (hunk 1 above)
Comment (Ploug) [inline Ball of Duty Server/Program.cs:50]: We dont want to split for each space, we want the whole string after the first word.

Ie GetCommand("ban player with id 12") would return "player with id 12". not "player", "with", "id", "12"

**Pull request closed without merging**